### DISCUSSION TITLE
How is a jdeploy app updated? Can the app check for updates on start?

### DISCUSSION THREAD

**Discussion by jpage4500** [shannah/jdeploy#144] [Q&A] [answered]
Hi,

I'm very new to jdeploy.. just found out about it yesterday. But, I'm loving everything so far as it's exactly what I was looking for -- a way to make deploying a Java Swing app as easy as possible.

One thing I noticed in the developer docs is this:
>> Applications deployed using jDeploy can also receive updates automatically as they become available, so you can be assured that your users will always be working with the latest version of your application

How does this work? I don't see anything about it but I'm probably just not looking in the right place.

I previously had a 'check for update' logic in my app which would look at the github releases and compare. It was great at detecting there was a new version available -- but couldn't actually download or apply the update.

I see the option in the installer but wasn't sure when or how often this actually works? What happens when an update is detected? Does it matter if I'm using npm or github as the destination? I still haven't decided which of these (or both?) to use..

<img width="493" alt="image" src="https://github.com/user-attachments/assets/6cb46fb4-f7ce-4e71-a835-93b3ff35d87b">

Also, I was considering still leaving my update check in the app.. if an update is found, is there a way to kick out the update process from within the app? Or, worst-case I could just show a dialog asking the user to paste a command into a terminal

FWIW here's my project.. I've got a few other questions but don't want to ask too many in a single post :)

https://github.com/jpage4500/AndroidDeviceManager

thanks
joe

**Comment by shannah** [comment, marked as answer]
> How does this work? I don't see anything about it but I'm probably just not looking in the right place.
On application launch, it will check the "source" of the app (either npm or github releases, depending on how it was deployed), and check to see if there is a newer version.  If so, it will download the new version, and then proceed with launching the updated app.

Right now you can deploy via npm or github, but I'd like to add other options, such as self hosting soon.

There is also a feature request to specify the auto update settings for the app, and not even give the user an option to enable/disable them.
